pydis_site/apps/content/resources/guides/python-guides/subclassing_context.md

```diff
@@ -0,0 +1,116 @@
+subclass, it will look like this:
+
+```python
+from typing import Optional
+
+from discord import RawReactionActionEvent
+from discord.ext import commands
+
+
+class CustomContext(commands.Context):
+    async def prompt(
+            self,
+            message: str,
+            *,
+            timeout=30.0,
+            delete_after=True
+    ) -> Optional[bool]:
+        """Prompt the author with an interactive confirmation message.
+
+        This method will send the `message` content, and wait for max `timeout` seconds
+        (default is `30`) for the author to react to the message.
+
+        If `delete_after` is `True`, the message will be deleted before returning a
+        boolean or None (if the author didn't respond) indicating whether the author
+        confirmed or denied.
+        """
+        msg = await self.send(message)
+
+        for reaction in ('\N{WHITE HEAVY CHECK MARK}', '\N{CROSS MARK}'):
+            await msg.add_reaction(reaction)
+
+        confirmation = None
+
+        def check(payload: RawReactionActionEvent):
+            # 'nonlocal' works almost like 'global' except for functions inside of
+            # functions. This means that when 'confirmation' is changed, that will
+            # apply to the variable above
+            nonlocal confirmation
+
+            if payload.message_id != msg.id or payload.user_id != self.author.id:
+                return False
+
+            emoji = str(payload.emoji)
+
+            if emoji == '\N{WHITE HEAVY CHECK MARK}':
+                confirmation = True
+                return True
+
+            elif emoji == '\N{CROSS MARK}':
+                confirmation = False
+                return True
+
+            # This means that it was neither of the two emojis added, so the author
+            # added some other unrelated reaction.
+            return False
+
+        try:
+            await self.bot.wait_for('raw_reaction_add', check=check, timeout=timeout)
+        except asyncio.TimeoutError:
+            # The 'confirmation' variable is still None in this case
+            pass
+
+        if delete_after:
+            await msg.delete()
+
+        return confirmation
+```
+
+After creating your context subclass, you need to override the `get_context()` method on your
+Bot class and change the default of the `cls` parameter to this subclass:
+
+```python
+from discord.ext import commands
+
+
+class CustomBot(commands.Bot):
+    async def get_context(self, message, *, cls=CustomContext):  # From the above codeblock
+        return await super().get_context(message, cls=cls)
+```
+
+Now that discord.py is using your custom context, you can use it in a command. For example:
+
+```python
+from discord.ext import commands
+
+
+bot = CustomBot(...)  # Replace with the arguments for the bot
+
+
+@bot.command()
+async def massban(ctx: CustomContext, members: commands.Greedy[discord.Member]):
+    if not await ctx.prompt(f"Are you sure you want to ban {len(members)}?"):
```

Comment:
I think it would be better if we were to extract this part out so it's clearer.
Also the prompt should have "members" at the end, other wise it just looks like:
`Are you sure you want to ban 5?` which doesn't make much sense.

```suggestion
    prompt = await ctx.prompt(f"Are you sure you want to ban {len(members)} members?")
    if not prompt:
```